Code/MinimalLib/minilib.cpp

```diff
@@ -556,71 +486,62 @@ std::string JSMol::condense_abbreviations_from_defs(
+  Abbreviations::condenseMolAbbreviations(get(), abbrevs, maxCoverage);
+  return mappingToJsonArray(get());
 }
 
-std::string JSMol::generate_aligned_coords(const JSMol &templateMol,
+std::string JSMolBase::generate_aligned_coords(const JSMolBase &templateMol,
                                            const std::string &details) {
-  assert(d_mol && templateMol.d_mol);
-  if (!templateMol.d_mol->getNumConformers()) {
+  if (!templateMol.get().getNumConformers()) {
     return "";
   }
-  return MinimalLib::generate_aligned_coords(*d_mol, *templateMol.d_mol,
+  return MinimalLib::generate_aligned_coords(get(), templateMol.get(),
                                              details.c_str());
 }
 
-int JSMol::has_coords() const {
-  assert(d_mol);
-  if (!d_mol->getNumConformers()) {
+int JSMolBase::has_coords() const {
+  if (!get().getNumConformers()) {
     return 0;
   }
-  return (d_mol->getConformer().is3D() ? 3 : 2);
+  return (get().getConformer().is3D() ? 3 : 2);
 }
 
-double JSMol::normalize_depiction(int canonicalize, double scaleFactor) {
-  assert(d_mol);
-  if (!d_mol->getNumConformers()) {
+double JSMolBase::normalize_depiction(int canonicalize, double scaleFactor) {
+  if (!get().getNumConformers()) {
     return -1.;
   }
-  return RDDepict::normalizeDepiction(*d_mol, -1, canonicalize, scaleFactor);
+  return RDDepict::normalizeDepiction(get(), -1, canonicalize, scaleFactor);
 }
 
-void JSMol::straighten_depiction(bool minimizeRotation) {
-  assert(d_mol);
-  if (!d_mol->getNumConformers()) {
+void JSMolBase::straighten_depiction(bool minimizeRotation) {
+  if (!get().getNumConformers()) {
     return;
   }
-  RDDepict::straightenDepiction(*d_mol, -1, minimizeRotation);
+  RDDepict::straightenDepiction(get(), -1, minimizeRotation);
 }
 
-bool JSMol::is_valid() const { return true; }
+bool JSMolBase::is_valid() const { return true; }
 
-std::pair<JSMolList *, std::string> JSMol::get_frags(
-    const std::string &details_json) {
-  assert(d_mol);
+std::pair<JSMolList *, std::string> JSMolBase::get_frags(
+    const std::string &details_json) const {
   std::vector<int> frags;
   std::vector<std::vector<int>> fragsMolAtomMapping;
   bool sanitizeFrags = true;
   bool copyConformers = true;
   MinimalLib::get_mol_frags_details(details_json, sanitizeFrags,
                                     copyConformers);
-  auto molFrags = MolOps::getMolFrags(*d_mol, sanitizeFrags, &frags,
+  auto molFrags = MolOps::getMolFrags(get(), sanitizeFrags, &frags,
                                       &fragsMolAtomMapping, copyConformers);
   return std::make_pair(
       new JSMolList(molFrags),
       MinimalLib::get_mol_frags_mappings(frags, fragsMolAtomMapping));
 }
 
-unsigned int JSMol::get_num_atoms(bool heavyOnly) const {
-  assert(d_mol);
-  return heavyOnly ? d_mol->getNumHeavyAtoms() : d_mol->getNumAtoms();
+unsigned int JSMolBase::get_num_atoms(bool heavyOnly) const {
+  return heavyOnly ? get().getNumHeavyAtoms() : get().getNumAtoms();
```

Comment:
oh, look, you snuck a use of the ternary operator in at some point in past without me noticing! 
;-)

Reply:
too late :)

Reply:
I have made the change. After the change the compiler correctly complains about the `JSMolBase::get_stereo_tags()` method, which used to be declared as `const` but is not actually `const` as it calls `MolOps::assignStereochemistry` and `CIPLabeler::assignCIPLabels`, both of which modify the underlying `d_mol`. Interestingly, the compiler was perfectly happy modifying `d_mol` inside a `const` method when `MolOps::assignStereochemistry` and `CIPLabeler::assignCIPLabels` were being directly passed `*d_mol` as parameter. Oh, well. Anyway, I removed the `const` qualifier from `JSMolBase::get_stereo_tags`.